mne/decoding/search_light.py

```diff
@@ -222,17 +231,24 @@ def score(self, X, y):
         score : array, shape (n_samples, n_estimators)
             Score for each estimator / data slice couple.
         """
+        from sklearn.metrics import make_scorer
+
         self._check_Xy(X)
         if X.shape[-1] != len(self.estimators_):
             raise ValueError('The number of estimators does not match '
                              'X.shape[-1]')
+        # If scoring is None (default), the predictions are internally
+        # generated by estimator.score(). Else, we must first get the
+        # predictions based on the scorer.
+        self.scoring = (make_scorer(self.scoring) if self.scoring is not None
+                        else self.scoring)
```

Comment:
object parameters should not be change.

You could store the scorer in `self.scorer_`, but I think we need not store the scorer, WDYT @agramfort ?